src/backend/base/langflow/components/models/IBMWatsonxModel.py

```diff
@@ -0,0 +1,89 @@
+from pydantic.v1 import SecretStr
+
+from langchain_ibm import ChatWatsonx
+
+from langflow.base.models.model import LCModelComponent
+from langflow.field_typing import LanguageModel
+from langflow.inputs import DropdownInput, FloatInput, IntInput, SecretStrInput, MessageTextInput
+
+
+class IBMWatsonXComponent(LCModelComponent):
+    display_name = "IBM watsonx"
+    description = "Generate text using IBM watsonx."
+    icon = "IBM"
+    name = "IBMwatsonxModel"
+    inputs = LCModelComponent._base_inputs + [
+        DropdownInput(
+            name="model",
+            display_name="Model",
+            info="The name of the model to use.",
+            options=["ibm/granite-13b-chat-v2"],
+            value="ibm/granite-13b-chat-v2",
+        ),
+        DropdownInput(
+            name="url",
+            display_name="URL",
+            info="https://ibm.github.io/watsonx-ai-python-sdk/setup_cloud.html#authentication.",
+            options=[
+                "https://us-south.ml.cloud.ibm.com",
+                "https://eu-gb.ml.cloud.ibm.com",
+                "https://eu-de.ml.cloud.ibm.com",
+                "https://jp-tok.ml.cloud.ibm.com",
+            ],
+            value="https://us-south.ml.cloud.ibm.com",
+        ),
+        MessageTextInput(
+            name="project_id",
+            display_name="Project_ID",
+            info="https://www.ibm.com/docs/en/watsonx-as-a-service?topic=projects.",
+        ),
+        SecretStrInput(
+            name="watsonx_api_key",
+            display_name="Watsonx API Key",
+            info="The Watson API Key to use for Watsonx.",
+        ),
+        IntInput(
+            name="max_output_tokens",
+            display_name="Max Output Tokens",
+            info="The maximum number of tokens to generate.",
+        ),
+        FloatInput(
+            name="top_p",
+            display_name="Top P",
+            info="The maximum cumulative probability of tokens to consider when sampling.",
+            advanced=True,
+        ),
+        FloatInput(name="temperature", display_name="Temperature", value=0.1),
+        IntInput(
+            name="top_k",
+            display_name="Top K",
+            info="Decode using top-k sampling: consider the set of top_k most probable tokens. Must be positive.",
```

Comment:
if `k` is non-positive, does `ChatWatsonx` throw a helpful error? Or should you specifically check this param before building